packages/server/src/sdk/app/oauth2/utils.ts

```diff
@@ -62,24 +62,32 @@ const trackUsage = async (id: string) => {
   })
 }
 
-// TODO: check if caching is worth
 export async function getToken(id: string) {
-  const config = await get(id)
-  if (!config) {
-    throw new HttpError(`oAuth config ${id} count not be found`)
-  }
-
-  const resp = await fetchToken(config)
-
-  const jsonResponse = await resp.json()
-  if (!resp.ok) {
-    const message = jsonResponse.error_description ?? resp.statusText
-
-    throw new Error(`Error fetching oauth2 token: ${message}`)
-  }
+  const token = await cache.withCacheWithDynamicTTL(
+    cache.CacheKey.OAUTH2_TOKEN(id),
+    async () => {
+      const config = await get(id)
+      if (!config) {
+        throw new HttpError(`oAuth config ${id} count not be found`)
+      }
+
+      const resp = await fetchToken(config)
+
+      const jsonResponse = await resp.json()
+      if (!resp.ok) {
```

Comment:
I think you probably want to check `resp.ok` _before_ grabbing `resp.json()`, in case it fails in such a way that it doesn't return JSON. Then when formatting the error, call `resp.text()` which is less likely to throw.

Reply:
This should cover controlled errors. `401` or `403` and `error_description` is part of the oauth2 spec, so we want to extract it if provided. Otherwise, the `resp.statusText` should get other issues, such as connection ones